File: TextosJapones/src/main/java/org/jisho/textosJapones/database/dao/VincularDao.java
package org.jisho.textosJapones.database.dao;

import org.jisho.textosJapones.model.entities.Vinculo;
import org.jisho.textosJapones.model.entities.mangaextractor.MangaTabela;
import org.jisho.textosJapones.model.entities.mangaextractor.MangaVinculo;
import org.jisho.textosJapones.model.enums.Language;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.util.List;
import java.util.UUID;

public interface VincularDao {

	void update(String base, Vinculo obj) throws ExcessaoBd;

	Vinculo select(String base, UUID id) throws ExcessaoBd;
	
	Vinculo select(String base, Integer volume, String mangaOriginal, Language original, String arquivoOriginal,
			String mangaVinculado, Language vinculado, String arquivoVinculado) throws ExcessaoBd;

	Vinculo select(String base, Integer volume, String mangaOriginal, String arquivoOriginal, String mangaVinculado,
			String arquivoVinculado) throws ExcessaoBd;

	Vinculo select(String base, Integer volume, String mangaOriginal, Language original, String mangaVinculado,
			Language vinculado) throws ExcessaoBd;

	void delete(String base, Vinculo obj) throws ExcessaoBd;

	UUID insert(String base, Vinculo obj) throws ExcessaoBd;

	Boolean createTabelas(String nome) throws ExcessaoBd;

	List<String> getMangas(String base, Language linguagem) throws ExcessaoBd;

	List<String> getTabelas() throws ExcessaoBd;

	List<MangaVinculo> selectVinculo(String base, String manga, Integer volume, Float capitulo, Language linguagem) throws ExcessaoBd;

	List<MangaTabela> selectTabelasJson(String base, String manga, Integer volume, Float capitulo, Language linguagem) throws ExcessaoBd;

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/database/dao/VincularDao.kt
package br.com.fenix.processatexto.database.dao

import br.com.fenix.processatexto.model.entities.mangaextractor.MangaTabela
import br.com.fenix.processatexto.model.entities.mangaextractor.MangaVinculo
import br.com.fenix.processatexto.model.entities.processatexto.Vinculo
import br.com.fenix.processatexto.model.enums.Language
import java.sql.SQLException
import java.util.*


interface VincularDao : RepositoryDao<UUID?, Vinculo> {
    @Throws(SQLException::class)
    fun update(base: String, obj: Vinculo)

    @Throws(SQLException::class)
    fun select(base: String, id: UUID): Optional<Vinculo>

    @Throws(SQLException::class)
    fun select(
        base: String,
        volume: Int,
        mangaOriginal: String,
        original: Language?,
        arquivoOriginal: String,
        mangaVinculado: String,
        vinculado: Language?,
        arquivoVinculado: String
    ): Optional<Vinculo>

    @Throws(SQLException::class)
    fun select(base: String, volume: Int, mangaOriginal: String, arquivoOriginal: String, mangaVinculado: String, arquivoVinculado: String): Optional<Vinculo>

    @Throws(SQLException::class)
    fun select(base: String, volume: Int, mangaOriginal: String, original: Language, mangaVinculado: String, vinculado: Language): Optional<Vinculo>

    @Throws(SQLException::class)
    fun delete(base: String, obj: Vinculo)

    @Throws(SQLException::class)
    fun insert(base: String, obj: Vinculo): UUID

    @Throws(SQLException::class)
    fun createTabelas(nome: String): Boolean

    @Throws(SQLException::class)
    fun getMangas(base: String, linguagem: Language): List<String>

    @get:Throws(SQLException::class)
    val tabelas: List<String>

    @Throws(SQLException::class)
    fun selectVinculo(base: String, manga: String, volume: Int, capitulo: Float, linguagem: Language): MutableList<MangaVinculo>

    @Throws(SQLException::class)
    fun selectTabelasJson(base: String, manga: String, volume: Int, capitulo: Float, linguagem: Language): MutableList<MangaTabela>
}